experiments/Atmos_LES/bomex.jl

```diff
@@ -0,0 +1,503 @@
+function atmos_source!(s::BomexSponge, atmos::AtmosModel, source::Vars, state::Vars, diffusive::Vars, aux::Vars, t::Real)
+
+  z_max = s.z_max
+  z_sponge = s.z_sponge
+  α_max = s.α_max
+  γ = s.γ
+  u_geostrophic = s.u_geostrophic
+  u_slope       = s.u_slope
+  v_geostrophic = s.v_geostrophic
+
+  z          = altitude(atmos.orientation,aux)
+  u_geo      = SVector(u_geostrophic + u_slope * z, v_geostrophic, 0)
+  ẑ          = vertical_unit_vector(atmos.orientation, aux)
+  # Accumulate sources
+  if z_sponge <= z
+    r = (z - z_sponge)/(z_max-z_sponge)
+    β_sponge = α_max * sinpi(r/2)^s.γ
+    source.ρu -= β_sponge * (state.ρu .- state.ρ * u_geo)
+  end
+  return nothing
+end
+
+"""
+  BomexTendencies (Source)
+Moisture, Temperature and Subsidence tendencies
+"""
+struct BomexTendencies{FT} <: Source
+  "Advection tendency in total moisture `[s⁻¹]`"
+  ∂qt∂t_peak::FT
+  "Lower extent of piecewise profile (moisture term) `[m]`"
+  zl_moisture::FT   
+  "Upper extent of piecewise profile (moisture term) `[m]`"
+  zh_moisture::FT
+  "Cooling rate `[K/s]`"
+  ∂θ∂t::FT
+  "Lower extent of piecewise profile (subsidence term) `[m]`"
+  zl_sub::FT
+  "Upper extent of piecewise profile (subsidence term) `[m]`"
+  zh_sub::FT
+  "Subsidence peak velocity"
+  w_sub::FT
+  "Max height in domain"
+  z_max::FT
+end
+function atmos_source!(s::BomexTendencies, atmos::AtmosModel, source::Vars, state::Vars, diffusive::Vars, aux::Vars, t::Real)
+  FT = eltype(state)
+  ρ     = state.ρ
+  z     = altitude(atmos.orientation,aux)
+  q_tot = state.moisture.ρq_tot / state.ρ
+  TS    = thermo_state(atmos.moisture, atmos.orientation, state, aux)
+  
+  # Moisture tendencey (sink term) 
+  # Temperature tendency (Radiative cooling)
+  # Large scale subsidence
+  # Unpack struct
+  zl_moisture = s.zl_moisture
+  zh_moisture = s.zh_moisture
+  z_max       = s.z_max
+  zl_sub      = s.zl_sub
+  zh_sub      = s.zh_sub
+  w_sub       = s.w_sub
+  ∂qt∂t_peak  = s.∂qt∂t_peak
+  ∂θ∂t        = s.∂θ∂t
+  k̂           = vertical_unit_vector(atmos.orientation, aux)
+
+  # Piecewise profile for advective moisture forcing
+  P     = air_pressure(TS)
+  ∂T∂t_peak  = air_temperature_from_liquid_ice_pottemp_given_pressure(∂θ∂t, P, PhasePartition(q_tot))
+
+  # Thermodynamic state identification
+  q_pt  = PhasePartition(TS)
+  cvm   = cv_m(TS)
+  
+  linscale_moisture = (z-zl_moisture)/(zh_moisture-zl_moisture)
+  linscale_temp = (z-zl_sub) / (z_max-zl_sub)
+  linscale_sub = (z-zl_sub) / (zh_sub-zl_sub)
+
+  if z <= zl_moisture
+    ρ∂qt∂t = ρ*∂qt∂t_peak
+  elseif zl_moisture < z <= zh_moisture
+    ρ∂qt∂t = ρ*(∂qt∂t_peak - ∂qt∂t_peak * linscale_moisture)
+  else
+    ρ∂qt∂t = -zero(FT)
+  end
+  
+  if z <= zl_sub
+    ρ∂T∂t = ρ*∂T∂t_peak
+  elseif  zh_sub < z <= z_max
+    ρ∂T∂t = ρ*∂T∂t_peak*linscale_temp
+  else
+    ρ∂T∂t = -zero(FT)
+  end
+  
+  # Moisture tendency 
+  source.moisture.ρq_tot += ρ∂qt∂t
+  
+  # Temperature tendency
+  source.ρe += cvm*ρ∂T∂t + e_int_v0*ρ∂qt∂t
```

Comment:
Make changes @szy21 points out, then
```suggestion
  source.ρe += cvm*exner(T, ρ, q)*ρ∂\theta∂t + e_int_v0*ρ∂qt∂t
```

Reply:
Hmm, I thought I had made this change a while back , let me check the commit versions that I've pushed.

Reply:
This should be fixed in the current version.